beaker/client/api_providers.py

```diff
@@ -0,0 +1,130 @@
+from enum import Enum
+from abc import ABC, abstractmethod
+from algosdk.v2client.algod import AlgodClient
+from algosdk.v2client.indexer import IndexerClient
+
+
+class Network(Enum):
+    """Provides consistent way to reference the most common network options"""
+
+    MainNet = "MainNet"
+    TestNet = "TestNet"
+    BetaNet = "BetaNet"
+    SandNet = "SandNet"
+
+
+class APIProvider(ABC):
+    """abstract class to provide interface for API providers"""
+
+    def __init__(self, network: Network):
```

Comment:
I feel like it might make more sense to have the dev initialize this class directly while passing in the api provider/network as an arg (opposed to initializing different subclasses). For example 

```py
api = APIProvider(AlgoNode.MainNet)
```

Reply:
I don't quite understand this?

Reply:
My bias against OOP might be showing but I think in general if having one class that takes in an argument accomplishes the same thing as multiple subclasses. 

For example

```py
algonode_mainnet = APIProvider(AlgoNode.MainNet)
purestake_testnet = APIProvider(PureStake.TestNet)
```
vs

```py
algonode_mainnet = AlgoNode(Network.MainNet)
purestake_testnet = PureStake(Network.TestNet)
```

Of course, this assumes we can write a single `algod` and `indexer` method under `APIProvider` as discussed above